starknet_py/net/account/account_client.py

```diff
@@ -179,6 +192,64 @@ async def add_transaction(
 
         return await super().add_transaction(await self._prepare_invoke_function(tx))
 
+    async def declare(
+        self,
+        compilation_source: Optional[StarknetCompilationSource] = None,
+        compiled_contract: Optional[str] = None,
+        max_fee: Optional[int] = 0,
+        version: Optional[int] = 0,
+        search_paths: Optional[List[str]] = None,
+    ) -> dict:
+        if not compiled_contract and not compilation_source:
+            raise ValueError(
+                "One of compiled_contract or compilation_source is required."
+            )
+
+        if not compiled_contract:
+            compiled_contract = Compiler(
+                contract_source=compilation_source, cairo_path=search_paths
+            ).compile_contract()
+        contract_class = create_contract_class(compiled_contract)
+
+        res = await self.add_transaction(
+            tx=Declare(
+                contract_class=contract_class,
+                sender_address=self.address,
+                max_fee=max_fee,
+                signature=[],
+                nonce=0,
+                version=version,
+            )
+        )
+
+        if res["code"] != StarkErrorCode.TRANSACTION_RECEIVED.name:
+            raise Exception("Transaction not received")
+
+        return res["class_hash"]
+
+    async def deploy(
+        self,
+        class_hash: str,
+        constructor_calldata: List[int],
+        salt: Optional[int] = None,
+        version: Optional[int] = 0,
+    ) -> AddressRepresentation:
+        res = await self.add_transaction(
+            tx=Deploy(
+                contract_address_salt=ContractAddressSalt.get_random_value()
+                if salt is None
+                else salt,
+                contract_definition=class_hash,  # TODO: for now it is wrong
+                constructor_calldata=constructor_calldata,
```

Comment:
One thing that came to my mind now: To still support translating calldata between python and starknet, we need to have an ABI for the given class_hash, and use the same data transformation logic as we used in deploy before. To implement this we will need to first support #179.

cc @Solpatium